File: 3_March/Mar_23/weather.js
const axios = require('axios');

// Your OpenWeatherMap API key
const apiKey = 'YOUR_API_KEY'; // Replace this with your actual API key

// Function to fetch weather data
async function getWeather(city) {
    try {
        const response = await axios.get(`http://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${apiKey}&units=metric`);
        const weatherData = response.data;
        return weatherData;
    } catch (error) {
        console.error('Error fetching weather:', error);
        return null;
    }
}

// Function to display weather information
function displayWeather(weather) {
    if (!weather) {
        console.log('Weather data not available.');
        return;
    }

    console.log(`Weather in ${weather.name}:`);
    console.log(`Temperature: ${weather.main.temp}°C`);
    console.log(`Description: ${weather.weather[0].description}`);
    console.log(`Humidity: ${weather.main.humidity}%`);
    console.log(`Wind Speed: ${weather.wind.speed} m/s`);
}

// Main function
async function main() {
    const city = 'London'; // Change this to the desired city
    const weather = await getWeather(city);
    displayWeather(weather);
}

main();

File: 1_January/Jan_16/weather.py
import requests
import json

# Replace 'YOUR_API_KEY' with your actual OpenWeatherMap API key
api_key = 'YOUR_API_KEY'

def get_weather(city_name):
    base_url = f'http://api.openweathermap.org/data/2.5/weather?q={city_name}&appid={api_key}&units=metric'
    
    response = requests.get(base_url)
    data = response.json()

    if data["cod"] != "404":
        main_data = data["main"]
        temperature = main_data["temp"]
        humidity = main_data["humidity"]
        weather_info = data["weather"][0]["description"]

        print(f"Weather in {city_name}:")
        print(f"Temperature: {temperature}°C")
        print(f"Humidity: {humidity}%")
        print(f"Description: {weather_info}")
    else:
        print("City not found.")

if __name__ == '__main__':
    city_name = input("Enter city name: ")
    get_weather(city_name)
